test/src/test_CloudColor.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+   Copyright (c) 2019 Arduino.  All rights reserved.
+*/
+
+/**************************************************************************************
+   INCLUDE
+ **************************************************************************************/
+
+#include <catch.hpp>
+
+#include <TestUtil.h>
+#include <ArduinoCloudThing.h>
+
+/**************************************************************************************
+  TEST CODE
+ **************************************************************************************/
+
+/************************************************************************************/
+SCENARIO("Arduino Cloud Properties ", "[ArduinoCloudThing::CloudColor]") {
+  WHEN("Set invalid color HSB") {
+    GIVEN("CloudProtocol::V2") {
+
+
+      CloudColor color_test = CloudColor(0.0, 0.0, 0.0);
+
+      Color value_color_test = color_test.getValue();
+      REQUIRE(value_color_test.setColorHSB(500.0, 20.0, 30.0) == false);
+
+    }
+  }
+
+  WHEN("Set and Get different RGB colors") {
+    GIVEN("CloudProtocol::V2") {
+
+      bool verify;
+      uint8_t r, g, b;
+
+      CloudColor color_test = CloudColor(0.0, 0.0, 0.0);
+
+      Color value_color_test = color_test.getValue();
+
+      value_color_test.setColorRGB(128, 64, 64);
+      value_color_test.getRGB(r, g, b);
+      verify = r == 128 && g == 64 && b == 64;
+
+      REQUIRE(verify);
```

Comment:
A better idea might be to split the verification into three separate statements, e.g.
```suggestion
REQUIRE(r == 128);
REQUIRE(g == 64);
REQUIRE(b == 64);
```
This way the unit test framework will produce a meaningful error message in case of test failure, something like `TEST FAIL, r should be 128 but is 127 instead` intstead of `TEST FAIL, verfiy is false, should be true`.

I'm mentioning it once here, but it applies everywhere within the change.

Reply:
Done